src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UploadDecisionLetterHandler.java

```diff
@@ -69,20 +70,28 @@ public boolean canHandle(
                 .map(Optional::get)
                 .collect(Collectors.toList());
 
-
         Optional<List<IdValue<DocumentWithMetadata>>> maybeExistingLegalRepDocuments =
             asylumCase.read(LEGAL_REPRESENTATIVE_DOCUMENTS);
 
         final List<IdValue<DocumentWithMetadata>> existingLegalRepDocuments =
             maybeExistingLegalRepDocuments.orElse(emptyList());
 
+        for (IdValue<DocumentWithMetadata> existingLegalRepDocument : existingLegalRepDocuments) {
+
+            if (existingLegalRepDocument.getValue().getTag().equals(DocumentTag.HO_DECISION_LETTER)) {
```

Comment:
Tag could be null I think in other legal rep documents across the journey

Reply:
@makowalski when looking at this line here:

(https://github.com/hmcts/ia-case-api/blob/28bafe93346709e5e80301faa7ed178d09a3d2e1/src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithMetadata.java#L63)

Is is possible to check for a null tag?

Reply:
It is in getter, I am not sure of the purpose of this check to be honest. You will get NPE throws from the requireNonNull(tag) method when you put tag=null in the constructor and then try to getTag()

Reply:
I have this check after the for loop now: 

`if (existingLegalRepDocument.getValue().getTag() != null) {`

Will this be suitable or leave as is?